### DISCUSSION TITLE
How to write a table in Pdf using PyMupdf

### DISCUSSION THREAD

**Discussion by manish59** [pymupdf/PyMuPDF#1288] [Q&A]
Hi, 

Is there a way to generate a pdf with table, as Im trying to generate an invoice out of it. 

Thanks,
Manis

**Comment by JorjMcKie** [comment]
(Py-) MuPDF does not use the feature that PDF supports for this. And most table-like things you encounter in PDFs are not created like that either.
But it is easy to define a table in PyMuPDF and fill text or whatever in the cells. Attached is an example that draws a 4 x 10 table to a PDF page, fills each with some rotated text and marks that text with text marker annotations.
Hope you can use it as a starting point.
[markers.zip](https://github.com/pymupdf/PyMuPDF/files/7216344/markers.zip)
Page looks like this:
![grafik](https://user-images.githubusercontent.com/8290722/134484043-23fa102e-cbe8-47ba-904b-22ab8bf329b8.png)
